// @parcel/core/lib/InternalConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createConfig = createConfig;
exports.addDevDependency = addDevDependency;

function createConfig({
  isSource,
  searchPath,
  env,
  result,
  includedFiles,
  invalidateOnFileCreate,
  devDeps,
  shouldRehydrate,
  shouldReload,
  shouldInvalidateOnStartup
}) {
  return {
    isSource,
    searchPath,
    env,
    result: result !== null && result !== void 0 ? result : null,
    resultHash: null,
    includedFiles: includedFiles !== null && includedFiles !== void 0 ? includedFiles : new Set(),
    invalidateOnFileCreate: invalidateOnFileCreate !== null && invalidateOnFileCreate !== void 0 ? invalidateOnFileCreate : [],
    pkg: null,
    pkgFilePath: null,
    devDeps: devDeps !== null && devDeps !== void 0 ? devDeps : new Map(),
    shouldRehydrate: shouldRehydrate !== null && shouldRehydrate !== void 0 ? shouldRehydrate : false,
    shouldReload: shouldReload !== null && shouldReload !== void 0 ? shouldReload : false,
    shouldInvalidateOnStartup: shouldInvalidateOnStartup !== null && shouldInvalidateOnStartup !== void 0 ? shouldInvalidateOnStartup : false
  };
}

function addDevDependency(config, name, version) {
  config.devDeps.set(name, version);
}
